### PR TITLE
tool/gocross: add ts_macext build tag for Xcode builds

### PR DESCRIPTION
**Pull request by mihaip**
It's used to control various opt-in functionality for the macOS and iOS apps, and was lost in the migration to gocross.

Updates tailscale/tailscale#7769

### CODE REVIEW

**Comment by mihaip** [comment]
Would appreciate tips for how to try this out (as far as what I have to tweak on the corp side to build with this change). Relatedly, do we need to do anything beyond a normal OSS update to pick up the change once it's merged?

File: tool/gocross/autoflags.go [hunk 1]
```diff
@@ -102,6 +102,10 @@ func autoflagsForTest(argv []string, env *Environment, goroot, nativeGOOS, nativ
 		cgo = nativeGOOS == "darwin"
 		tags = append(tags, "omitidna", "omitpemdecrypt")
 		if env.IsSet("XCODE_VERSION_ACTUAL") {
+			// If we're building via Xcode, we must be making the extension
+			// version (as opposed to tailscaled on Mac).
+			tags = append(tags, "ts_macext")
```
Comment (danderson) [inline tool/gocross/autoflags.go:107]: Does ts_macext cover the App Store build as well?

**Review by danderson — APPROVED** [review]
Oops, sorry for missing that!

File: tool/gocross/autoflags.go (hunk 1 above)
Comment (mihaip) [reply tool/gocross/autoflags.go:107]: As far as I can tell it was specified for all the Xcode builds (iOS, macOS, macsys): https://github.com/tailscale/corp/blob/72401f9d3bd6eecfde5f5a2cb70a5469fb012bf0/xcode/ipn-go-bridge/default.a.do#L43-L51.

**Pull request merged**